### ISSUE TITLE
OSError: [Errno 6] No such device or address: '/dev/tty'

### ISSUE THREAD

**Issue by berho** [dbcli/mycli#899]
I want to run commands from a file in a cron job and get the following error:

```
Traceback (most recent call last):
  File "/usr/bin/mycli", line 11, in <module>
    load_entry_point('mycli==1.16.0', 'console_scripts', 'console')()
  File "/usr/lib/python3/dist-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/share/mycli/mycli/main.py", line 1059, in cli
    sys.stdin = open('/dev/tty')
OSError: [Errno 6] No such device or address: '/dev/tty'
```

when I run the same command in terminal no error occures!
`mycli  -d MyConnection < add-data.sql >>add-data.log 2>&1`

**Comment by rolandwalker** [comment]
Hi!

That's a reasonable thing to want, and I think we can fix it so that `mycli` doesn't attempt to open `/dev/tty` if the `--no-warn` option is given.

**Comment by pasenor** [comment]
released in version 1.23.0
